cupy/core/_routines_statistics.pyx

```diff
@@ -173,6 +177,89 @@ __device__ min_max_st<T> my_argmax_complex(
     if (is_nan(b.value.imag())) return b;
     return (a.value >= b.value) ? a : b;
 }
+template <typename T>
+__device__ min_max_st<T> my_nanargmax(
+        const min_max_st<T>& a, const min_max_st<T>& b) {
+    if (a.index == -1) return b;
+    if (b.index == -1) return a;
+    if (a.value == b.value && !is_nan(a.value) && !is_nan(b.value))
+        return min_max_st<T>(a.value, min(a.index, b.index));
+    if (!is_nan(a.value) && !is_nan(b.value) && a.value >= b.value) return a;
+    if (!is_nan(b.value) && !is_nan(a.value) && b.value >= a.value) return b;
+    if (!is_nan(a.value) && is_nan(b.value)) return a;
+    if (!is_nan(b.value) && is_nan(a.value)) return b;
+    return min_max_st<T>(-1, -1);
```

Comment:
When all values in the given array are NaN, it will return `-1`. Is it expected behavior?